PowerEditor/src/functionList.xml

```diff
@@ -123,21 +123,21 @@
 						<nameExpr expr="\w+\Z" />
 					</className>
 					<function
-						mainExpr="^\s*((static|public|protected|private|final)*(\s+(static|public|protected|private|final))+\s+)?(function\s+)+(\w+(\s+\w+)?(\s+|\*\s+|\s+\*|\s+\*\s+))?(\w+\s*::)?(?!(if|while|for|switch))[\w~]+\s*\([^\{]*\{"
+						mainExpr="^\s*((static|public|protected|private|final)*(\s+(static|public|protected|private|final))+\s+)?(function\s+)+(\w+(\s+\w+)?(\s+|\*\s+|\s+\*|\s+\*\s+))?(\w+\s*::)?(?!(if|while|for|switch|foreach))[\w~]+\s*\([^\{]*\{"

```

Comment:
(yet) outside the scope of this PR, but I guess the `(\s+|\*\s+|\s+\*|\s+\*\s+)` could be shortened to `(\s+)`, as there can't be such `*` in PHP. Probably the rule has been copied from another language.

Reply:
Same with the double colon `::`, that can occur in static call but not in function definition.